presto-orc/src/main/java/com/facebook/presto/orc/Filters.java

```diff
@@ -140,6 +244,120 @@ int staticScore()
         }
     }
 
+    public static class FloatRange
+            extends Filter
+    {
+        private final float lower;
+        private final boolean lowerUnbounded;
+        private final boolean lowerExclusive;
+        private final float upper;
+        private final boolean upperUnbounded;
+        private final boolean upperExclusive;
+
+        FloatRange(float lower, boolean lowerUnbounded, boolean lowerExclusive, float upper, boolean upperUnbounded, boolean upperExclusive, boolean nullAllowed)
+        {
+            super(nullAllowed);
+            this.lower = lower;
+            this.lowerUnbounded = lowerUnbounded;
+            this.lowerExclusive = lowerExclusive;
+            this.upper = upper;
+            this.upperUnbounded = upperUnbounded;
+            this.upperExclusive = upperExclusive;
+        }
+
+        @Override
+        public boolean testFloat(float value)
+        {
+            if (!lowerUnbounded) {
```

Comment:
@tdcmeehan Since float time is approximate, couldn't this cause issues?

Reply:
Yeah.  I guess exclusivity would mean b - <<first discreet value before b>>, perhaps `Math.nextUp` would work?

Reply:
@tdcmeehan Tim, I'm reluctant to make this change because I don't know how to make sure it is safe. If it is not, the resulting bugs will be super hard to track down.